Code_Exercises/CMakeLists.txt

```diff
@@ -8,32 +8,102 @@
-  target_compile_definitions("${prefix}_${source}" PUBLIC
-    -DSYCL_LANGUAGE_VERSION=2020)
-  if(${SYCL_ACADEMY_USE_COMPUTECPP})
-    target_compile_definitions("${prefix}_${source}" PUBLIC -DSYCL_ACADEMY_USE_COMPUTECPP)
+message(STATUS ">> CMake Exercises")
+message(STATUS "SYCL_ACADEMY_BUILD_EXERCISES ${SYCL_ACADEMY_BUILD_EXERCISES}")
+
+# CMake function to build a ComputeCPP configuration and create executable binaries
+# from ComputeCPP libraries. 
+function( add_sycl_executable_computecpp prefix source )
+  if(NOT DEFINED ${CMAKE_BUILD_TYPE})
+    set(CMAKE_BUILD_TYPE "DEBUG")
   endif()
+
+  add_executable("${prefix}_${source}" "${source}.cpp")
+  target_compile_definitions("${prefix}_${source}" PUBLIC -DSYCL_LANGUAGE_VERSION=2020)
+  target_compile_definitions("${prefix}_${source}" PUBLIC -DSYCL_ACADEMY_USE_COMPUTECPP)
   target_include_directories("${prefix}_${source}" PRIVATE
-    ${PROJECT_SOURCE_DIR}/Utilities/include ${PROJECT_SOURCE_DIR}/External/stb)
-  target_link_libraries("${prefix}_${source}" PRIVATE Threads::Threads)
+    ${PROJECT_SOURCE_DIR}/Utilities/include 
+    ${PROJECT_SOURCE_DIR}/External/stb)
+  target_link_libraries("${prefix}_${source}" PRIVATE Threads::Threads) 
   target_link_libraries("${prefix}_${source}" PUBLIC Catch2::Catch2)
-  set_target_properties("${prefix}_${source}" PROPERTIES CXX_STANDARD 17
-    CXX_STANDARD_REQUIRED ON)
-
-  add_sycl_to_target(
-    TARGET "${prefix}_${source}"
-    SOURCES "${source}.cpp"
-  )
-    
+  set_target_properties("${prefix}_${source}" PROPERTIES CXX_STANDARD 17 CXX_STANDARD_REQUIRED ON)  
   add_test(${prefix}_${source} ${prefix}_${source})
 endfunction()
 
-function(add_sycl_executable_dpcpp prefix source)
-  # Not implemented
+# CMake function to build a hipSYCL configuration and create executable binaries
+# from hipSYCL libraries. 
+function( add_sycl_executable_hipSYCL prefix source )
+  if(NOT DEFINED ${CMAKE_BUILD_TYPE})
+	  set(CMAKE_BUILD_TYPE "DEBUG")
+  endif()
+
+  add_executable("${prefix}_${source}" "${source}.cpp")
+  target_compile_definitions("${prefix}_${source}" PUBLIC -DSYCL_LANGUAGE_VERSION=2020)
+  target_include_directories("${prefix}_${source}" PRIVATE
+    ${PROJECT_SOURCE_DIR}/Utilities/include 
+    ${PROJECT_SOURCE_DIR}/External/stb)
+  target_link_libraries("${prefix}_${source}" PRIVATE Threads::Threads) 
+  target_link_libraries("${prefix}_${source}" PUBLIC Catch2::Catch2)
+  set_target_properties("${prefix}_${source}" PROPERTIES CXX_STANDARD 17 CXX_STANDARD_REQUIRED ON)  
+endfunction()
+
+# CMake function to build a DPC++ configuration and create executable binaries
+# from oneAPI libraries. 
+function( add_sycl_executable_dpcpp prefix source )
+  if(NOT DEFINED ${CMAKE_BUILD_TYPE})
+    set(CMAKE_BUILD_TYPE "DEBUG")
+  endif()
+
+  set(CMAKE_VERBOSE_MAKEFILE ON)
+  set(TARGET_NAME "${prefix}_${source}")
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_COMPILER "icpx")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsycl")
```

Comment:
These variables should all have been set before we reach here (either by the user, or by the find module, or by the set_target_properties for the CXX_VERSION.

Reply:
Removed. Done.